### PR TITLE
ViewModel Tests

### LINKED ISSUE #53: Test Cases
Placeholder for project part 3 submission to remind us to have test cases (UI, unit, integration)

Write runnable tests for your model and control classes. Provide intent tests for the requirements you have done. Deliver the test code to your source repository. If you have test data files, also include those. Test data should be realistic.

### PR DESCRIPTION
**Pull request by armiantos**
# Summary
- Give example on how to use Mockito and Roboelectric to unit test

# TODO
- Finish other view models

# Question
- LibraryViewModel is tightly coupled with the UI (it get's values from UI) is it okay if I refactor part of the viewmodel to make it look closer to an API? (Create addBook function, create deleteBook function, etc.)
  - From group discussion, seems like everyone's ok with it so will be proceeding with refactor

Contributes to #53

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -47,8 +47,12 @@ dependencies {
     // Optional -- Robolectric environment
     testImplementation 'androidx.test:core:1.3.0'
     // Optional -- Mockito framework
-    androidTestImplementation 'org.mockito:mockito-core:3.2.4'
-    androidTestImplementation 'org.mockito:mockito-android:3.2.4'
+    testImplementation 'org.mockito:mockito-core:3.6.0'
```
Comment (armiantos) [inline app/build.gradle:50]: `testImplementation` is meant for unit tests and `androidTestImplementation` is meant for instrumented tests

**Review by armiantos — COMMENTED** [review]
Guidelines to keep in mind while reviewing this PR

File: app/src/test/java/com/example/unlibrary/unlibrary/UnlibraryViewModelTest.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
+package com.example.unlibrary.unlibrary;
```
Comment (cyrusdiego) [inline app/src/test/java/com/example/unlibrary/unlibrary/UnlibraryViewModelTest.java:1]: nit file header

File: app/src/test/java/com/example/unlibrary/unlibrary/UnlibraryViewModelTest.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(Toast.class)
+public class UnlibraryViewModelTest {
```
Comment (cyrusdiego) [inline app/src/test/java/com/example/unlibrary/unlibrary/UnlibraryViewModelTest.java:38]: nit: i can't remember what the TA said about documentation but I think in Golnoush's PR we at least do class headers so just to be consistent please add a header for the class

**Review by cyrusdiego — COMMENTED** [review]
just left some nits, overall this looks good and will be a good reference for the other viewmodels, ping me for approval :P

File: app/src/test/java/com/example/unlibrary/unlibrary/UnlibraryViewModelTest.java (hunk 3 above)
Comment (armiantos) [reply app/src/test/java/com/example/unlibrary/unlibrary/UnlibraryViewModelTest.java:38]: Will do!

File: app/src/test/java/com/example/unlibrary/unlibrary/UnlibraryViewModelTest.java (hunk 2 above)
Comment (armiantos) [reply app/src/test/java/com/example/unlibrary/unlibrary/UnlibraryViewModelTest.java:1]: Added!

File: app/src/test/java/com/example/unlibrary/unlibrary/UnlibraryViewModelTest.java (hunk 3 above)
Comment (armiantos) [reply app/src/test/java/com/example/unlibrary/unlibrary/UnlibraryViewModelTest.java:50]: Added!

**Review by cyrusdiego — APPROVED** [review]
LGTM! good work on this :)

**Comment by armiantos** [comment]
Sweet, thank you!

**Pull request merged**